core/src/main/java/greencity/security/filters/AccessTokenAuthenticationFilter.java

```diff
@@ -24,35 +28,42 @@
 public class AccessTokenAuthenticationFilter extends OncePerRequestFilter {
     private final JwtTool jwtTool;
     private final AuthenticationManager authenticationManager;
+    private final UserService userService;
 
     /**
      * Constructor.
      */
-    public AccessTokenAuthenticationFilter(JwtTool jwtTool, AuthenticationManager authenticationManager) {
+    public AccessTokenAuthenticationFilter(JwtTool jwtTool, AuthenticationManager authenticationManager,
+                                           UserService userService) {
         this.jwtTool = jwtTool;
         this.authenticationManager = authenticationManager;
+        this.userService = userService;
     }
 
     /**
      * Checks if request has token in header, if this token still valid, and set
      * authentication for spring.
      *
-     * @param request this is servlet that take request
+     * @param request  this is servlet that take request
      * @param response this is response servlet
-     * @param chain this is filter of chain
+     * @param chain    this is filter of chain
      */
     @Override
     public void doFilterInternal(@SuppressWarnings("NullableProblems") HttpServletRequest request,
                                  @SuppressWarnings("NullableProblems") HttpServletResponse response,
                                  @SuppressWarnings("NullableProblems") FilterChain chain)
         throws IOException, ServletException {
         String token = jwtTool.getTokenFromHttpServletRequest(request);
+
         if (token != null) {
             try {
                 Authentication authentication = authenticationManager
                     .authenticate(new UsernamePasswordAuthenticationToken(token, null));
-                log.info("User successfully authenticate - {}", authentication.getPrincipal());
-                SecurityContextHolder.getContext().setAuthentication(authentication);
+                User user = userService.findByEmail((String) authentication.getPrincipal());
+                if (user.getUserStatus() != UserStatus.DEACTIVATED) {
```

Comment:
My first idea was that this new if statement is not covered by unit test. Have you measured the test coverage for this class?

Reply:
But then I've got a better idea. You don't have to load the user and then check the status. You should create a new method in `UserService` that will query the database for a user "by email and status equals not deactivated".

Reply:
Another problem with the current approach is that if user is not found, a NPE will be thrown.

Reply:
Done!